src/Collections/PropertyCollection.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+
+namespace ArmadilloEstates\Collections;
+
+
+use ArmadilloEstates\Entities\PropertyEntity;
+use ArmadilloEstates\Interfaces\IPropertyCollection;
+
+class PropertyCollection implements IPropertyCollection
+{
+    private $allProperties;
+
+    public function __construct(array $properties)
+    {
+        $this->propertyCheck($properties);
+        $this->allProperties = $properties;
+    }
+
+    private function propertyCheck($properties)
+    {
+        foreach ($properties as $property) {
+            if (!($property instanceof PropertyEntity)) {
```

Comment:
this is essentially a typehint, so at the moment you are typehinting to a class, which breaks dependency inversion